src/OrchardCore.Modules/OrchardCore.Autoroute/Routing/AutorouteValuesAddressScheme.cs

```diff
@@ -37,7 +39,17 @@ public IEnumerable<Endpoint> FindEndpoints(RouteValuesAddress address)
             return [];
         }
 
-        (var found, var autorouteEntry) = _entries.TryGetEntryByContentItemIdAsync(contentItemId).GetAwaiter().GetResult();
+        var task = _entries.TryGetEntryByContentItemIdAsync(contentItemId);
+
+        bool found;
+        AutorouteEntry autorouteEntry;
+
+        if (!task.IsCompletedSuccessfully)
+        {
+            task.GetAwaiter().GetResult(); // Wait for the task to complete if it hasn't already.
+        }
+
+        (found, autorouteEntry) = task.Result;
```

Comment:
What's the trick here, why is this better? Also for `SitemapValuesAddressScheme`.